subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/install/internal/AdoptOpenJdkDownloader.java

```diff
@@ -69,11 +69,17 @@ public String getShortDisplayName() {
         return getTransport(source).getRepository().withProgressLogging().resource(resourceName);
     }
 
-    private void downloadResource(URI source, File tmpFile, ExternalResource resource) {
-        resource.withContent(inputStream -> {
-            LOGGER.info("Downloading {} to {}", resource.getDisplayName(), tmpFile);
-            copyIntoFile(source, inputStream, tmpFile);
-        });
+    private void downloadResource(URI source, File targetFile, ExternalResource resource) {
+        final File downloadFile = new File(targetFile.getAbsoluteFile() + ".part");
+        try {
+            resource.withContent(inputStream -> {
+                LOGGER.info("Downloading {} to {}", resource.getDisplayName(), targetFile);
+                copyIntoFile(source, inputStream, downloadFile);
+            });
+            downloadFile.renameTo(targetFile);
```

Comment:
We should not ignore the return value here.
I also believe that using `java.nio.file.Files#move` with the right options might be a better choice.